SignalServiceKit/src/Storage/OWSStorage.m

```diff
@@ -184,21 +183,15 @@ - (YapDatabaseConnection *)newConnection
 
 - (YapDatabaseConnection *)registrationConnection
 {
-    @synchronized(self)
-    {
-        if (!self.registrationConnectionCached) {
-            YapDatabaseConnection *connection = [super registrationConnection];
+    YapDatabaseConnection *connection = [super registrationConnection];
```

Comment:
Backup exports previously needed to block until the backup storage was completely closed.  I did a bunch of work to support this, including this bit.